src/coreclr/jit/morph.cpp

```diff
@@ -7869,6 +7869,12 @@ GenTree* Compiler::fgMorphCall(GenTreeCall* call)
         optMethodFlags |= OMF_NEEDS_GCPOLLS;
     }
 
+    if (fgGlobalMorph && IsStaticHelperEligibleForExpansion(call))
+    {
+        // Current method has potential candidates for fgExpandStaticInit phase
+        setMethodHasStaticInit();
```

Comment:
curious, why don't we set this in `importer.cpp` itself?

Reply:
Like I mentioned in your PR, in importer we might only set it for inlinee's compiler so we have 3 options:
1) Do `impInlineRoot()->setMethodHasStaticInit()` -- will work but it is too conservative if we reject inlining in the middle - we'll have to discard this flag
2) copy flag to caller's Compiler object at some point
3) Just do that in morph

I decided to do that in morph. The biggest advantage in my case that in imported I had to do that in multiple places while here it's a single place